sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/implementation/KeyVaultCredentialPolicy.java

```diff
@@ -67,16 +72,20 @@ private static Map<String, String> extractChallengeAttributes(String authenticat
             return Collections.emptyMap();
         }
 
-        authenticateHeader =
-            authenticateHeader.toLowerCase(Locale.ROOT).replace(authChallengePrefix.toLowerCase(Locale.ROOT), "");
-
-        String[] attributes = authenticateHeader.split(", ");
+        String[] attributes = authenticateHeader.substring(authChallengePrefix.length()).split(", ");
         Map<String, String> attributeMap = new HashMap<>();
 
         for (String pair : attributes) {
-            String[] keyValue = pair.split("=");
+            // This is ugly, but we need to trim here because currently the 'claims' attribute comes after two spaces.
+            pair = pair.trim();
 
-            attributeMap.put(keyValue[0].replaceAll("\"", ""), keyValue[1].replaceAll("\"", ""));
+            if (pair.startsWith("claims=")) {
```

Comment:
When an authentication challenge includes `claims` we need to base64-decode them, which is why they're treated differently than other parameters. This also means we need to remove the quotes before decoding happens. That said, we could certainly run the replace code just for that particular pair.

Reply:
But doesn't the base64-decoding happen later one? Right now, the code here is

- pair.startsWith("claims=")
  - attributeMap.put("claims", pair.substring("claims".length()).replaceAll("\"",""));
- else
  - kvp[] split = pair.split("=");
  - attributeMap.put(kvp[0].replaceAll("\"",""), kvp[1].replaceAll("\"",""));

Which ends up being effectively the same code. I think this can all become the same code with the change to using `pair.split("=", 2)`, because if as you mention the claims value is base64 it was likely running into issues if the base64 value ended with the `=` or `==` fillers where those would have been dropped and became invalid base64.

Reply:
The content of the claims value is a base64-encoded string which can be padded with `=` characters at the end, so it can mess with splitting the string using that character.